crates/typst/src/math/equation.rs

```diff
@@ -76,6 +77,30 @@ pub struct EquationElem {
     #[borrowed]
     pub numbering: Option<Numbering>,
 
+    /// The alignment of the equation numbering.
+    ///
+    /// By default, the number is put at the `{end}` of the equation block. Both
+    /// `{start}` and `{end}` respects text direction; for absolute positioning,
+    /// use `{left}` or `{right}`.
+    ///
+    /// ```example
+    /// #set math.equation(numbering: "(1)", number-align: start)
+    ///
+    /// With natural units, we know:
+    /// $ E^2 = m^2 + p^2 $
+    /// ```
+    #[default(HAlignment::End)]
+    #[parse({
+        let option: Option<Spanned<HAlignment>> = args.named("number-align")?;
+        if let Some(Spanned { v: align, span }) = option {
+            if align == HAlignment::Center {
+                bail!(span, "equation number cannot be `center`-aligned");
+            }
+        }
+        option.map(|spanned| spanned.v)
+    })]
+    pub number_align: HAlignment,
```

Comment:
Let's wait until the tables PR is merged (shouldn't take long) and then this can be migrated to `OuterHAlignment`: https://github.com/typst/typst/blob/207a3f4f5cebe3e2cfef2b31c52474f09aed7961/crates/typst/src/layout/align.rs#L324-L333